From cf4f11dae2b086303a35bd2c9be38b4049eb1673 Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Wed, 10 Jan 2018 21:43:31 +0100
Subject: [PATCH] src/mmc_linux.c: Include sysmacros.h

libtool: link: x86_64-pc-linux-gnu-gcc -Wall -D_FILE_OFFSET_BITS=64 -pthread -march=native -mtune=native -O2 -pipe -Wl,-O1 -Wl,--hash-style=gnu -Wl,--sort-common -o fwup fwup-archive_open.o fwup-block_cache.o fwup-cfgfile.o fwup-cfgprint.o fwup-crc32.o fwup-eval_math.o fwup-fatfs.o fwup-functions.o fwup-fwfile.o fwup-fwup_apply.o fwup-fwup.o fwup-fwup_create.o fwup-fwup_list.o fwup-fwup_sign.o fwup-fwup_verify.o fwup-fwup_metadata.o fwup-fwup_genkeys.o fwup-mbr.o fwup-mmc_bsd.o fwup-mmc_linux.o fwup-mmc_osx.o fwup-mmc_windows.o fwup-pad_to_block_writer.o fwup-progress.o fwup-requirement.o fwup-resources.o fwup-simple_string.o fwup-sparse_file.o fwup-uboot_env.o fwup-util.o ../3rdparty/fwup-base64.o ../3rdparty/semver.c/fwup-semver.o ../3rdparty/fatfs/src/fwup-ff.o ../3rdparty/fatfs/src/option/fwup-unicode.o  -Wl,--as-needed -lconfuse -larchive -lsodium -pthread
fwup-mmc_linux.o: In function `mmc_is_path_at_device_offset':
mmc_linux.c:(.text+0x478): undefined reference to `minor'
mmc_linux.c:(.text+0x485): undefined reference to `major'
collect2: error: ld returned 1 exit status
---
 src/mmc_linux.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/mmc_linux.c b/src/mmc_linux.c
index d1a09ff..50f224b 100644
--- a/src/mmc_linux.c
+++ b/src/mmc_linux.c
@@ -25,6 +25,7 @@
 #include <sys/mount.h>
 #include <sys/stat.h>
 #include <sys/wait.h>
+#include <sys/sysmacros.h>
 
 #include <fcntl.h>
 #include <inttypes.h>
-- 
2.16.0.rc1

